sync/statesync/trie_sync_stats.go

```diff
@@ -128,15 +128,17 @@ func (t *trieSyncStats) updateETA(sinceUpdate time.Duration, now time.Time) {
 		// provide a separate ETA for the account trie syncing step since we
 		// don't know the total number of storage tries yet.
 		log.Info("state sync: syncing account trie", "ETA", roundETA(leafsTime))
-		return
+		return leafsTime
 	}
 
-	triesTime := now.Sub(t.triesStartTime) * time.Duration(t.triesRemaining) / time.Duration(t.triesSynced)
+	triesTime := now.Sub(t.triesStartTime) * time.Duration(t.triesRemaining/t.triesSynced)
```

Comment:
I'm not sure that this is the right calculation for triesTime.
```golang
total-time-so-far = now.Sub(t.triesStartTime)
```
during that time, we have completed `(t.triesSynced-t.triesRemaining)` tries.
that means that the time per trie is
```golang
time-per-trie = total-time-so-far /  (t.triesSynced-t.triesRemaining)
```
and the remaining time is `t.triesRemaining * time-per-trie`.

expanding that:
```golang
triesTime = t.triesRemaining * now.Sub(t.triesStartTime) / (t.triesSynced-t.triesRemaining)
```

Reply:
> during that time, we have completed (t.triesSynced-t.triesRemaining) tries.
we've completed `t.triesSynced` tries.

Reply:
ohh - so, in that case
```golang
triesTime := (now.Sub(t.triesStartTime) * time.Duration(t.triesRemaining)) / time.Duration(t.triesSynced)
```
since we want to get some reasonable precision.. otherwise, `t.triesRemaining/t.triesSynced` would be very lossy.

Reply:
this re-introduces the overflow

Reply:
Can we implement this in a common location? [timer.EstimateETA](https://github.com/ava-labs/avalanchego/blob/master/utils/timer/eta.go#L23) seems like it can probably do this (and shouldn't have the overflow issue). If that method doesn't work for this can we make one that does?